python/pyiceberg/table/__init__.py

```diff
@@ -639,6 +639,12 @@ def __init__(
         self.length = length or data_file.file_size_in_bytes
 
 
+def _starmap_open_manifest(
```

Comment:
Same here, maybe better to let `_open_manifest` accept a single tuple argument, and then we can directly expand those.